source/index.html.md

```diff
@@ -51,6 +53,57 @@ Filmmakers expects the API key to be included in all API requests to the server
 You must replace <code>API_KEY</code> with your personal API key.
 </aside>
 
+## OAuth
+
+Filmmakers API supports the OAuth flow with refresh tokens.
+
+### Setting up OAuth Access
+
+Once you have been issued an API key with OAuth Authorization enabled, you can use the following configuration from your client code:
+
+[Filmmakers OpenID Configuration](https://www.filmmakers.eu/.well-known/openid-configuration)
+
+See example on how to fetch the OAuth Access Token with curl.
+
+```shell
+export CLIENT_ID="your-client-id-goes-here"
+export CLIENT_SECRET="your-client-secret-goes-here"
+curl -s -X POST "http://filmmakers.test/oauth/token" \
+           -H "Content-Type: application/x-www-form-urlencoded" \
+           --data-urlencode "grant_type=client_credentials" \
+           --data-urlencode "client_id=$CLIENT_ID" \
+           --data-urlencode "client_secret=$CLIENT_SECRET" \
+           --data-urlencode "scope=public"
+```
+
+```json
+{"access_token":"Uvm37RVcZano399REpMXh837fDT-jtSc0lTvmEexpKI","token_type":"Bearer","expires_in":7200,"scope":"public","created_at":1730119409}
+```
+
+<aside class="notice">
+You must replace <code>CLIENT_ID</code> and <code>CLIENT_SECRET</code> with your credentials.
+</aside>
+
+### Using OAuth Access Tokens
+
+Filmmakers expects the OAuth Access Token to be included in all API requests to the server in a header that looks like the following:
+
+`Authorization: Bearer ACCESS_TOKEN`
+
+OAuth Access Tokens expire in 120 minutes and have to be refreshed, your client software is responsible for automating the OAuth flow.
```

Comment:
```suggestion
OAuth access tokens expire after a set period (currently 120 minutes, though this may change in the future) and must be refreshed. Your client software is responsible for automating the OAuth flow. You can always retrieve the expiry time of an access token from the introspection endpoint listed in the [Filmmakers OpenID Configuration](https://www.filmmakers.eu/.well-known/openid-configuration)." 
```